### PR TITLE
Added query results limit when the search exceed 10000 hits

### LINKED ISSUE #6097: [Vulnerabilities dashboards] Search error when results exceed 10000 elements
| Wazuh | Rev  | Browser                      |
|-------|------|------------------------------|
| 4.8.0 | xyzw | Chrome, Firefox, Safari, etc |

**Description**

In the new Inventory table that uses the data-grid component when we make a search with results that exceed position 10000, the search throws an error.

Related issue: #5763

```
search_phase_execution_exception: [illegal_argument_exception] Reason: Result window is too large, from + size must be less than or equal to: [10000] but was [14940]. See the scroll api for a more efficient way to request large data sets. This limit can be set by changing the [index.max_result_window] index level setting.
```

image

## Tasks

- Add a toast error to notify the user that the search fails because exceeds the 10000 results
- Research possible search customization to make the search

[https://opensearch.org/blog/point-in-time/](https://opensearch.org/blog/point-in-time/)
[https://www.elastic.co/guide/en/elasticsearch/reference/7.10/paginate-search-results.html#paginate-search-results](https://www.elastic.co/guide/en/elasticsearch/reference/7.10/paginate-search-results.html#paginate-search-results)
[https://opensearch.org/docs/latest/im-plugin/index-settings/#dynamic-index-settings](https://opensearch.org/docs/latest/im-plugin/index-settings/#dynamic-index-settings)

### PR DESCRIPTION
**Pull request by Machi3mfl**
### Description
This PR, adds a callout showing a warning message when the inventory table exceeds the 10000 hits.

Closes #6097 

## Added hits count and warning tooltip 

Screenshot 2023-11-08 at 11 50 21

https://github.com/wazuh/wazuh-dashboard-plugins/assets/6089438/d610c133-91ab-4c54-a69e-63c28ceab665

## CSV export limited to 10000 rows

Screenshot 2023-11-08 at 12 13 50

### Testing

To test the solution it's necessary to create the `wazuh-states-vulnerabilities` index and inject docs:
https://github.com/wazuh/wazuh-dashboard-plugins/issues/5763#issuecomment-1779564103

### Check List
- [x] All tests pass
  - [x] `yarn test:jest`
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by lucianogorza** [comment]
### All hits
all_hits

**Review by lucianogorza — CHANGES_REQUESTED** [review]
## All hits 🟢 
all_hits

### Export CSV 🔴 
image

**Only export the last request (1000 hits in this case)**

image

### Filtered hits 🟢 
filtered_hits

### Export CSV 🔴 
**Only export the last request (405 hits in this case)**

image

**Comment by lucianogorza** [comment]
## Filtered hits
### Export CSV 🟢 
image

image

LGTM

File: plugins/main/public/components/overview/vulnerabilities/dashboards/inventory/inventory.tsx [hunk 1]
```diff
@@ -127,56 +129,69 @@ const InventoryVulsComponent = () => {
...
+                      tooltip={results?.hits?.total && results?.hits?.total > MAX_ENTRIES_PER_QUERY ? {
+                        ariaLabel: 'Warning',
+                        content: `The query results has exceeded the limit of 10,000 hits. To provide a better experience the table only shows the first ${formatNumWithCommas(MAX_ENTRIES_PER_QUERY)} hits.`,
+                        iconType: 'alert',
+                        position: 'top'
+                      } : undefined}
+                    />
+                    <EuiButtonEmpty
+                      disabled={results?.hits?.total === 0}
+                      size="xs"
```
Comment (lucianogorza) [inline plugins/main/public/components/overview/vulnerabilities/dashboards/inventory/inventory.tsx:164]: I suggest to use the prop "isLoading" with a state:
```
isLoading={isExporting}
```
And set a status in the function `onClickExportResults`:

```
    try {
      setIsExporting(true)
      await exportSearchToCSV(params);
    } catch (error) {
      const searchError = ErrorFactory.create(HttpError, { error, message: 'Error downloading csv report' })
      ErrorHandler.handleError(searchError);
    } finally {
      setIsExporting(false)
    }
```

**Review by lucianogorza — APPROVED** [review]

**Comment by lucianogorza** [comment]
### CR 🟢 
### Test 🟢

**Review by jbiset — APPROVED** [review]
## CR: :green_circle: 

## TEST :green_circle: 

### All Hits :green_circle: 

all_hits

### Filtered Hits :green_circle: 

filtered_hits

### Export All Hits :green_circle: 

export_all_hits

### Export Filtered Hits :green_circle:  

export_filtered_hits_1235

**Pull request merged**